daemons/controld/controld_control.c

```diff
@@ -209,20 +223,9 @@ crmd_exit(crm_exit_t exit_code)
 
 /* Clean up as much memory as possible for valgrind */
 
-    for (GList *iter = controld_globals.fsa_message_queue; iter != NULL;
-         iter = iter->next) {
-        fsa_data_t *fsa_data = (fsa_data_t *) iter->data;
-
-        crm_info("Dropping %s: [ state=%s cause=%s origin=%s ]",
-                 fsa_input2string(fsa_data->fsa_input),
-                 fsa_state2string(controld_globals.fsa_state),
-                 fsa_cause2string(fsa_data->fsa_cause), fsa_data->origin);
-        delete_fsa_input(fsa_data);
-    }
-
     controld_clear_fsa_input_flags(R_MEMBERSHIP);
 
-    g_list_free(controld_globals.fsa_message_queue);
+    g_list_free_full(controld_globals.fsa_message_queue, free_fsa_data);
```

Comment:
It doesn't really matter, but you can skip the function and just use `(GDestroyNotify) delete_fsa_input` here if you would like.  We are inconsistent about which we do, but fewer lines is better all else being equal as far as I'm concerned.

Reply:
I can't find a citation but Ken had wanted to avoid these types of casts, instead creating wrappers like this one where needed. We didn't go through and change all the existing casts but we used wrappers going forward.

With that being said I'm fine with casting, personally.

Reply:
Huh, I wonder why he wanted to do that.  I guess I don't really care one way or another, but if I were writing this code I would go with the cast just so there's less stuff to look at.